src/inmanta/deploy/scheduler.py

```diff
@@ -675,37 +703,83 @@ async def report_resource_state(
+                        }
+                        self._state.dirty.update(dependant_resources)
+                        for dependant in dependant_resources:
+                            # No point in re-trying dependants since this resource was skipped
+                            # FIXME as part of https://github.com/inmanta/inmanta-core/issues/8423
+                            self._timer_manager.remove_resource(dependant)
+
                 # propagate events
                 if details.attributes.get(const.RESOURCE_ATTRIBUTE_SEND_EVENTS, False):
-                    provides: Set[ResourceIdStr] = self._state.requires.provides_view().get(resource, set())
                     event_listeners: Set[ResourceIdStr] = {
                         dependant
                         for dependant in provides
                         if (dependant_details := self._state.resources.get(dependant, None)) is not None
                         # default to True for backward compatibility, i.e. not all resources have the field
                         if dependant_details.attributes.get(const.RESOURCE_ATTRIBUTE_RECEIVE_EVENTS, True)
                     }
-                    if event_listeners:
-                        # do not pass deploying tasks because for event propagation we really want to start a new one,
-                        # even if the current intent is already being deployed
-                        task = Deploy(resource=resource)
-                        assert task in self._work.agent_queues.in_progress
-                        priority = self._work.agent_queues.in_progress[task]
-                        self._work.deploy_with_context(
-                            event_listeners,
-                            reason=f"Deploying because an event was received from {resource}",
-                            priority=priority,
-                            deploying=set(),
-                        )
+                    concerned_resources.update(event_listeners)
+
+                if concerned_resources:
+                    # do not pass deploying tasks because for event propagation we really want to start a new one,
+                    # even if the current intent is already being deployed
+                    task = Deploy(resource=resource)
+                    assert task in self._work.agent_queues.in_progress
+                    priority = self._work.agent_queues.in_progress[task]
+                    self._work.deploy_with_context(
+                        concerned_resources,
+                        reason=f"Deploying because an event was received from {resource}",
+                        priority=priority,
+                        deploying=set(),
+                    )
+
+        # No matter the deployment result, schedule a re-deploy for this resource
+        self._timer_manager.update_resource(resource, dirty=is_dirty)
```

Comment:
Should also be under the lock. And perhaps we should simply read from `self._state.compliance_status` so we don't need `is_dirty`?